### PR TITLE
docs: initial upload of styles used by doc linter

### PR DESCRIPTION
**Pull request by bob-watson**
This PR replaces  PR #46854 and includes the edits to the comments from that PR.

Adds styles and configuration files for contributors to use the Vale documentation linter in their IDE while they edit angular.io topics.
This PR includes documentation for using the documentation linter in the README.md file.
This PR does not alter any angular.io or other published documentation.

This PR is going to cause problems with the inclusive language CI test because it includes files with all the prohibited words in its tests to look for those words in the documentation.

These seem to be only two files:

* aio/tools/doc-linter/dictionaries/en_us/en_US-web.dic
* aio/tools/doc-linter/styles/Angular/Angular_Inclusive.yml

### CODE REVIEW

File: aio/tools/doc-linter/README.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+### Install the Vale extension for Visual Studio Code
+
+This procedure is the same for macOS and Windows.
+To install Vale in a different IDE, see [Installation][ValeDocsValeCliInstallation].
+
+> **Note:** <br />
+> The style tests used by Vale are stored in your clone of the `angular/angular` repo.
+> For best results, use the `main` branch of the `angular/angular` repo when you set the path of `vale.ini` in the IDE extension settings.
+> While the style tests don't change frequently, if you reference a fork or another branch of that repo,
+> you might miss an update to the style tests if it falls behind the `angular/angular` repo.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:241]: Here to, you could style it as a note (if you wanted) with:

```suggestion
> **Note**
> The style tests used by Vale are stored in your clone of the `angular/angular` repo.
> For best results, use the `main` branch of the `angular/angular` repo when you set the path of `vale.ini` in the IDE extension settings.
> While the style tests don't change frequently, if you reference a fork or another branch of that repo,
> you might miss an update to the style tests if it falls behind the `angular/angular` repo.
```

File: aio/tools/doc-linter/README.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+### Install the Vale extension for Visual Studio Code
+
+This procedure is the same for macOS and Windows.
+To install Vale in a different IDE, see [Installation][ValeDocsValeCliInstallation].
+
+> **Note:** <br />
+> The style tests used by Vale are stored in your clone of the `angular/angular` repo.
+> For best results, use the `main` branch of the `angular/angular` repo when you set the path of `vale.ini` in the IDE extension settings.
+> While the style tests don't change frequently, if you reference a fork or another branch of that repo,
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:240]: This is potentially misleading/confusing. You can't really reference a specific branch in a local file path.

File: aio/tools/doc-linter/README.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+> you might miss an update to the style tests if it falls behind the `angular/angular` repo.
+
+<!-- vale Angular.WriteGood_Illusions = NO -->
+<!-- vale Angular.Angular_CliReference = NO -->
+
+**To install the Visual Studio Code Vale extension, in Visual Studio Code:**
+
+1.  Install the `Vale` Visual Studio Code extension.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:248]: Maybe mention that you need to go the Extensions pane first :smiley:
You can also put the link the extension's page on the VSCode marketplace: https://marketplace.visualstudio.com/items?itemName=errata-ai.vale-server
(You can install an extension from there directly, which is potentially easier.)

File: aio/tools/doc-linter/README.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    1.  Search for `Vale`.
+    2.  Install `Vale`.
+2.  Navigate to **Preferences > Settings**.
+    
+    > **NOTE** <br />
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:253]: Here too (if desired):

```suggestion
    > **Note**
```

File: aio/tools/doc-linter/README.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+    > On macOS, the **Preferences** menu item is located under the **Code** nav-bar.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:254]: FWIW, on Windows **Preferences** is under **File**.

File: aio/tools/doc-linter/README.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+2.  Ensure that Vale is available on your `PATH`. To do this:
+    1.  Get the path to the `vale` application to add as the value for **Vale > Vale CLI : Path**.
+    2.  In the command-line tool, run `where vale.exe`.
+        Save the returned path for use in the next procedure.
+
+    3.  If this command does not return a path, restart Windows and repeat this step.
+3.  Update the clone of the `angular` repo on your system.
+4.  In the command-line tool:
+    1.  Navigate to your `github` workspace.
+    2.  In your `github` workspace, navigate to the `angular\aio\tools\doc-linter` directory.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:225]: https://github.com/angular/angular/pull/46854#discussion_r923513658 still applies here :smiley:

**Review by gkalpak — CHANGES_REQUESTED** [review]
A couple of comments that probably need addressing and a few optional suggestions.

BTW, you can ensure VSCode will show the vale extension as recommended for the workspace when people open the `angular/angular` repo, by adding it in [.vscode/extensions.json][1] (or leave it commented out at the bottom of the list if it's only useful when working with docs inside the `aio/` directory):

```diff
   "recommendations": [
     "devondcarew.bazel-code",
     "gkalpak.aio-docs-utils",
     "ms-vscode.vscode-typescript-tslint-plugin",
     "xaver.clang-format",
     // The following extensions are useful when working on angular.io (i.e. inside the `aio/` directory).
     //"angular.ng-template",
     //"dbaeumer.vscode-eslint",
+    //"errata-ai.vale-server",
   ],
```

[1]: https://github.com/angular/angular/blob/d583f85701b0424496e263bf1cbc2c619869437e/.vscode/extensions.json

**Comment by bob-watson** [comment]
Thanks for the notes!
The latest commit should address all of your comments.

File: aio/tools/doc-linter/README.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+Vale evaluates a markdown \(`.md`\) file when you open it in the IDE and each time you save it. Vale doesn't review your document as you make changes in it.
+
+When Vale identifies a problem, it can take a simple word substitution or rewriting several sentences or paragraphs to fix it.
+Sometimes, you might need to take a step back from the text and consider other alternatives.
+
+To find the best way to fix a problem that Vale identified, think of your reader, be flexible, and keep an open mind.
+The documentation should be understood by a range of readers, some of whom might not have much experience with Angular or even web development in general.
+
+> Remember that when a developer is reading documentation, they are often doing so while thinking about other things, such as coding an Angular application.
+> Keeping the docs easy to read, lets the developers concentrate on their coding.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit includes this.

File: aio/tools/doc-linter/README.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+
+These are some tips to help you resolve the issues that the lint tool identifies and to help you make your docs easier to read.
+
+### Refer to the style guides
+
+The lint tool tests against the styles found in these style guides.
+
+> **NOTE**: <br />
+> Not everything mentioned in the style guides have a test.
+> Style guides and tests can change and evolve.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit includes this.

File: aio/tools/doc-linter/README.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+Most style tests include links to relevant sections in these documents for more information.
+
+*   [Angular documentation style guide][AioGuideDocsStyleGuide]
+*   [Google Developer Documentation Style Guide][GoogleDevelopersStyle]
+
+### Split up long sentences
+
+Generally, shorter sentences are easier to read than longer ones.
+Long sentences occur when you try to say too much at once.
+Long sentences, as well as the use of parentheses, semi-colons, or a words identified as *too-wordy*, generally require rethinking and rewriting.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+In your command-line tool:
+
+<!-- vale Angular.WriteGood_Illusions = NO -->
+<!-- vale Angular.Angular_CliReference = NO -->
+
+1.  Run `brew install vale`
+2.  Ensure that Vale is available on your `$PATH`. To do this:
+    1.  Get the path to the `vale` application to add as the value for **Vale > Vale CLI : Path**.
+    2.  In the command-line tool, run `which vale`.
+        Save the returned path for use in the next procedure.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+In your command-line tool:
+
+<!-- vale Angular.WriteGood_Illusions = NO -->
+<!-- vale Angular.Angular_CliReference = NO -->
+
+1.  Run `choco install vale`.
+2.  Ensure that Vale is available on your `PATH`. To do this:
+    1.  Get the path to the `vale` application to add as the value for **Vale > Vale CLI : Path**.
+    2.  In the command-line tool, run `where vale.exe`.
+        Save the returned path for use in the next procedure.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md (hunk 1 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks. 
I removed this note. It seemed to confuse more than help when I read it again (and again).

File: aio/tools/doc-linter/README.md (hunk 2 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: See previous reply.

File: aio/tools/doc-linter/README.md (hunk 3 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks.
This should be clearer in the current version, although i didn't include the link.

File: aio/tools/doc-linter/README.md (hunk 4 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md (hunk 5 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md (hunk 6 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+1.  Run `brew install vale`
+2.  Ensure that Vale is available on your `$PATH`. To do this:
+    1.  Get the path to the `vale` application to add as the value for **Vale > Vale CLI : Path**.
+    2.  In the command-line tool, run `which vale`.
+        Save the returned path for use in the next procedure.
+
+    3.  If this command does not return a path, restart macOS and repeat this step.
+3.  Update the clone of the `angular` repo on your system. To do this, in the command-line tool:
+    1.  Navigate to your `github` workspace.
+    2.  In your `github` workspace, navigate to the `angular/aio/tools/doc-linter` directory.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks! The most recent commit fixes this.

File: aio/tools/doc-linter/README.md [hunk 13]
```diff
@@ -161,25 +161,24 @@ For more information, or to install Vale in another IDE or on another operating
 
 Before you proceed, make sure that:
 
-1.  You have cloned the [`angular/angular` repo][GithubAngularAngular] on your system. You can work in a fork of the `angular/angular` repo, 
-    but it's easier if your Vale installation uses an up-to-date clone of the `main` branch of the `angular/angular` repo.
-2.  You have a development or writing environment for `angular.io` in the `angular/angular` repo and you are able to build the `angular.io` docs.
+1.  You have cloned the [`angular/angular` repo][GithubAngularAngular] in the `git` workspace on your system. You can work in a fork of the `angular/angular` repo,
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:164]: It sounds like there might be some confusion here. This sentence seems to imply that there is a `git` workspace inside of which repositories are cloned. This is not exactly the case.

You can clone a `git` repository anywhere on your file system. The process of cloning will create a directory. This directory is called the "workspace" (or "working directory") in the context of `git`. So, basically, each repository that you clone on your local system creates a `git` workspace (or working directory).

I think this [StackOverflow answer][1] does a good job explaining the concept of the `git` workspace.

##
So, based on the above, I'd keep the sentence as:

```
You have cloned the [`angular/angular` repo][GithubAngularAngular] on your system.
```

[1]: https://stackoverflow.com/questions/59441589/how-is-repository-different-from-workspace#answer-63523191

File: aio/tools/doc-linter/README.md [hunk 14]
```diff
@@ -189,18 +188,19 @@ In your command-line tool:
...
-    3.  If this command does not return a path, restart macOS and repeat this step.
-3.  Update the clone of the `angular` repo on your system. To do this, in the command-line tool:
-    1.  Navigate to your `github` workspace.
-    2.  In your `github` workspace, navigate to the `angular/aio/tools/doc-linter` directory.
-    3.  In the `angular/aio/tools/doc-linter` directory, confirm that you have the `vale.ini` file.
-    4.  Run `pwd` to get the path to the `vale.ini` file for use in the next procedure.
+    2.  If this command does not return a path, restart macOS and repeat this step.
+3.  Get the path to the configuration file, `vale.ini`, to use in the **Vale > Vale CLI : Config** setting.
+    1.  In the command-line tool, navigate to your `git` workspace.
+    2.  In your `git` workspace, navigate to your clone of the `angular/angular` repo.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:199]: What you refer to as "the cloned repo" here _is_ the `git` workspace. See my comment on line 164 for more details.

So, I would combine these two sentences in one and only mention either "the `git` workspace" or "the cloned repo" (both options are fine imo).

The same applies to the Windows instructions below.

**Review by gkalpak — APPROVED** [review]
Basically, one last issue regarding `git` workspace. Otherwise lgtm :tada:

##
FYI, while the commit messages follow the [commit message guidelines][1], having separate commits for addressing review comments is against the _spirit_ of the guidelines. I.e. all commits should be self-contained. For example, for this PR all changes could be part of one commit, sine they are all implementing the addition of the doc linter.

This is not terribly important for docs commits, because those do not end up in the auto-generated [changelog][2]. But it does affect the readability of the commit history and makes it more difficult to find out when a particular change was introduced (and why/what is the context). To be clear, it does to remove anything from our ability to read history and find out these things - it just makes it less easy.

BTW, there is currently a mechanism for adding new commits to address review feedback (which makes it easier for the reviewer) while at the same time ensuring that all changes will be combined into one commit when the PR is merged into the target branch: [fixup commits][3]
However, they are less straight-forward (especially if you are not using the git CLI or a git UI that supports this feature).

This is not necessarily a blocker. I just wanted to bring it up.
(BTW, happy to elaborate more on any points that are not clear.)

[1]: https://github.com/angular/angular/blob/3ca985269d3e961321075f0e7facd10ef335be4c/CONTRIBUTING.md#commit
[2]: https://github.com/angular/angular/blob/3ca985269d3e961321075f0e7facd10ef335be4c/CHANGELOG.md
[3]: https://github.com/angular/angular/blob/main/docs/FIXUP_COMMITS.md

File: aio/tools/doc-linter/README.md [hunk 15]
```diff
@@ -28,43 +27,43 @@ Be sure to specify that this is a linter issue in your description.
 
 ## Fix documentation problems identified by Vale
 
-Vale evaluates a markdown \(`.md`\) file when you open it in the IDE and each time you save it. Vale doesn't review your document as you make changes in it.
+Vale evaluates a markdown \(`.md`\) file when you open it in the IDE and each time you save it. Vale does not review your document as you make changes in it.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:30]: The contractions style is still under discussion/consideration/revision so, for now, it's disabled in the `vale.ini` config file.

File: aio/tools/doc-linter/README.md (hunk 14 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks for the SO explanation. That's a big help!

File: aio/tools/doc-linter/README.md (hunk 13 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md]: Thanks!

**Review by gkalpak — APPROVED** [review]
:shipit:

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit bfddfe5622b2cc7cf874f9c0e303aed8b7dbae00.

**Pull request closed without merging**